### PR TITLE
DocsLinter trim and refactor

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description

- trimmed down `DocsLinter` to only check remote links
	- skips `console.improbable.io`
- used by `gdk-for-unity-docs` pipeline
- also warns if link is directly to a markdown file

#### Tests

- removed unused tests
- made sure link checker tests still work

### CODE REVIEW

File: tools/DocsLinter/LinkCheckCommand.cs [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+        ///     Side effects: Prints to the console.
+        /// </summary>
+        /// <param name="markdownFilePath">The fully qualified path of the Markdown file to check</param>
+        /// <param name="markdownFileContents">An object representing the Markdown file to check.</param>
+        /// <returns>A bool indicating the success of the check.</returns>
+        private bool CheckMarkdownFile(string markdownFilePath, SimplifiedMarkdownDoc markdownFileContents)
+        {
+            var allLinksValid = true;
+
+            foreach (var remoteLink in markdownFileContents.Links)
```
Comment (jamiebrynes7) [inline tools/DocsLinter/LinkCheckCommand.cs]: Nifty LINQ oneliner- 

```
return markdownFileContents.Links.All(link => CheckRemoteLink(markdownFilePath, link));
```

File: tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs [hunk 2]
```diff
@@ -8,33 +8,6 @@ namespace DocsLinter.Tests
...
-            var linkInline = new LinkInline("https://www.google.com", "Title");
-            var link = SimplifiedMarkdownDoc.ParseLink(linkInline);
-
-            Assert.IsInstanceOf<RemoteLink>(link);
-        }
-
-        [Test]
-        public void ParseLink_should_return_RemoteLink_when_url_starts_with_www()
-        {
-            var linkInline = new LinkInline("www.google.com", "Title");
```
Comment (jamiebrynes7) [inline tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs:23]: These tests are probably still valuable right? To ensure we catch various types of links

File: tools/DocsLinter/SimplifiedMarkdownDoc.cs [hunk 3]
```diff
@@ -24,39 +22,24 @@ public SimplifiedMarkdownDoc()
         /// <param name="markdownDoc"></param>
         public SimplifiedMarkdownDoc(MarkdownDocument markdownDoc)
         {
-            Links.AddRange(markdownDoc.Descendants().OfType<LinkInline>().Select(ParseLink));
-            Headings.AddRange(markdownDoc.Descendants().OfType<HeadingBlock>().Select(heading => new Heading(heading)));
+            Links.AddRange(markdownDoc.Descendants()
+                .OfType<LinkInline>()
+                .Where(inline => inline.Url.StartsWith("http") || inline.Url.StartsWith("www"))
```
Comment (zeroZshadow) [inline tools/DocsLinter/SimplifiedMarkdownDoc.cs]: This will miss URL's that don't start with www

**Review by zeroZshadow — CHANGES_REQUESTED** [review]

**Review by zeroZshadow — APPROVED** [review]

File: tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs (hunk 2 above)
Comment (paulbalaji) [reply tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs:23]: check out the new tests

File: tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs [hunk 4]
```diff
@@ -8,13 +7,58 @@ namespace DocsLinter.Tests
...
+        {
+            var url = CreateRemoteLink("https://www.google.com");
+            var markdown = GetSimplifiedMarkdown(url);
+
+            Assert.IsTrue(markdown.Links.Count == 1);
+            Assert.IsTrue(markdown.Links.Contains(url));
+        }
+
+        [Test]
+        public void Link_starting_with_site_name_is_ignored()
```
Comment (jamiebrynes7) [inline tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs]: Did we decide to ignore these links then?

File: tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs [hunk 5]
```diff
@@ -8,13 +7,58 @@ namespace DocsLinter.Tests
...
+            var url = CreateRemoteLink("google.com");
+            var markdown = GetSimplifiedMarkdown(url);
+
+            Assert.IsTrue(markdown.Links.Count == 0);
+            Assert.IsFalse(markdown.Links.Contains(url));
+        }
+
         [Test]
-        public void RemoteLink_should_assign_the_url_correctly()
+        public void Link_starting_with_www_is_ignored()
```
Comment (jamiebrynes7) [inline tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs]: Same question here

File: tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs (hunk 4 above)
Comment (paulbalaji) [reply tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs]: it matches the behaviour of `Uri.IsWellFormedUriString` https://docs.microsoft.com/en-us/dotnet/api/system.uri.iswellformeduristring?view=netcore-2.2#System_Uri_IsWellFormedUriString_System_String_System_UriKind_

File: tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs (hunk 5 above)
Comment (paulbalaji) [reply tools/DocsLinter/Tests/SimplifiedMarkdownDocTests.cs]: same answer as above

**Review by jamiebrynes7 — APPROVED** [review]

**Review by zeroZshadow — APPROVED** [review]

**Review by jamiebrynes7 — APPROVED** [review]

**Pull request merged**